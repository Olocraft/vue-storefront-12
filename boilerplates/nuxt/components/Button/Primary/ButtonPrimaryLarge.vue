<template>
  <button
    v-focus
    :disabled="disabled"
    class="inline-flex items-center justify-center px-6 py-3 text-base font-medium text-white uppercase rounded-md cursor-pointer bg-primary-500 shadow-base hover:bg-primary-600 active:bg-primary-700 disabled:bg-gray-200 disabled:text-gray-500 disabled:opacity-50 font-body hover:shadow-medium disabled:shadow-none disabled:cursor-not-allowed outline-violet"
    v-on="$listeners"
  >
    <slot />
  </button>
</template>

<script>
import { focus } from '../../../utils/focus-directive.js';

export default {
  name: 'ButtonPrimaryLarge',
  directives: {
    focus
  },
  props: {
    disabled: {
      type: Boolean,
      default: false
    }
  }
};
</script>
