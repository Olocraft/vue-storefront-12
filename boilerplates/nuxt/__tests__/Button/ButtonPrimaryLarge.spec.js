import { shallowMount } from '@vue/test-utils';
import ButtonPrimaryLarge from '../../components/Button/Primary/ButtonPrimaryLarge.vue';

let wrapper;

describe('ButtonPrimaryLarge.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(ButtonPrimaryLarge);
  });
  it('render component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
