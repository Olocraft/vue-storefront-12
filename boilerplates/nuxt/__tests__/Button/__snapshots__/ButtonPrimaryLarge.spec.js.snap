// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimaryLarge.vue render component correctly 1`] = `<button class="inline-flex items-center justify-center px-6 py-3 text-base font-medium text-white uppercase rounded-md cursor-pointer bg-primary-500 shadow-base hover:bg-primary-600 active:bg-primary-700 disabled:bg-gray-200 disabled:text-gray-500 disabled:opacity-50 font-body hover:shadow-medium disabled:shadow-none disabled:cursor-not-allowed outline-violet"></button>`;
